### ISSUE TITLE
Samba drive on Windows show half of the correct size

### ISSUE THREAD

**Issue by laggykiller** [rclone/rclone#6733]
#### What is the problem you are having with rclone?
I have a samba drive with port 9445. Section of `rclone.conf`
```
[mysmbdrive]
type = smb
host = localhost
user = laggykiller
port = 9445
pass = <redacted>
```
`mysmbdrive` contains a share named `Library`. It is hosted by Arch Linux samba.

I mount the drive on Windows by  `rclone.exe mount mysmbdrive:/Library D:`

The disk capacity and free space should be double than it shows:

![image](https://user-images.githubusercontent.com/61652821/215488698-43900683-baed-42e4-85fb-138986c16868.png)

However note that file sizes of the files are correctly displayed.

#### What is your rclone version (output from `rclone version`)
```
rclone v1.62.0-beta.6692.6b17044f8
- os/version: Microsoft Windows 11 Education 22H2 (64 bit)
- os/kernel: 10.0.25201.1000 Build 25201.1000.1000 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.20rc3
- go/linking: static
- go/tags: cmount
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)
Windows 11, 64 bit

#### Which cloud storage system are you using? (e.g. Google Drive)
Samba

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)
 `rclone.exe mount mysmbdrive:/Library D:`

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)
```
2023/01/30 21:15:31 DEBUG : rclone: Version "v1.62.0-beta.6692.6b17044f8" starting with parameters ["C:\\ProgramData\\chocolatey\\bin\\rclone.exe" "-vv" "mount" "mysmbdrive:/Library" "D:"]
2023/01/30 21:15:31 DEBUG : Creating backend with remote "mysmbdrive:/Library"
2023/01/30 21:15:31 DEBUG : Using config file from "C:\\Users\\laggykiller\\AppData\\Roaming\\rclone\\rclone.conf"
2023/01/30 21:15:31 DEBUG : fs cache: renaming cache item "mysmbdrive:/Library" to be canonical "mysmbdrive:Library"
2023/01/30 21:15:31 INFO  : smb://laggykiller@localhost:9445/Library/: poll-interval is not supported by this remote
2023/01/30 21:15:31 DEBUG : Network mode mounting is disabled
2023/01/30 21:15:31 DEBUG : Mounting on "D:" ("mysmbdrive Library")
2023/01/30 21:15:31 DEBUG : smb://laggykiller@localhost:9445/Library/: Mounting with options: ["-o" "attr_timeout=1" "-o" "uid=-1" "-o" "gid=-1" "--FileSystemName=rclone" "-o" "volname=mysmbdrive Library"]
2023/01/30 21:15:31 DEBUG : smb://laggykiller@localhost:9445/Library/: Init:
2023/01/30 21:15:31 DEBUG : smb://laggykiller@localhost:9445/Library/: >Init:
2023/01/30 21:15:31 DEBUG : /: Statfs:
2023/01/30 21:15:32 DEBUG : /: >Statfs: stat={Bsize:4096 Frsize:4096 Blocks:968907957 Bfree:267710278 Bavail:267710278 Files:1000000000 Ffree:1000000000 Favail:0 Fsid:0 Flag:0 Namemax:255}, errc=0
2023/01/30 21:15:32 DEBUG : /: Getattr: fh=0xFFFFFFFFFFFFFFFF
2023/01/30 21:15:32 DEBUG : /: >Getattr: errc=0
2023/01/30 21:15:32 DEBUG : /: Readlink:
2023/01/30 21:15:32 DEBUG : /: >Readlink: linkPath="", errc=-40
2023/01/30 21:15:32 DEBUG : /: Getxattr: name="non-existant-a11ec902d22f4ec49003af15282d3b00"
2023/01/30 21:15:32 DEBUG : /: >Getxattr: errc=-40, value=""
The service rclone has been started.
2023/01/30 21:15:32 DEBUG : /: Getattr: fh=0xFFFFFFFFFFFFFFFF
2023/01/30 21:15:32 DEBUG : /: >Getattr: errc=0
2023/01/30 21:15:32 DEBUG : /: Getattr: fh=0xFFFFFFFFFFFFFFFF
2023/01/30 21:15:32 DEBUG : /: >Getattr: errc=0
2023/01/30 21:15:32 DEBUG : /: Opendir:
2023/01/30 21:15:32 DEBUG : /: OpenFile: flags=O_RDONLY, perm=-rwxrwxrwx
2023/01/30 21:15:32 DEBUG : /: >OpenFile: fd=/ (r), err=<nil>
2023/01/30 21:15:32 DEBUG : /: >Opendir: errc=0, fh=0x0
2023/01/30 21:15:32 DEBUG : /: Getpath: Getpath fh=0
2023/01/30 21:15:32 DEBUG : /: >Getpath: errc=0, normalisedPath=""
2023/01/30 21:15:32 DEBUG : /: Statfs:
2023/01/30 21:15:32 DEBUG : /: >Statfs: stat={Bsize:4096 Frsize:4096 Blocks:968907957 Bfree:267710278 Bavail:267710278 Files:1000000000 Ffree:1000000000 Favail:0 Fsid:0 Flag:0 Namemax:255}, errc=0
2023/01/30 21:15:32 DEBUG : /: Getattr: fh=0xFFFFFFFFFFFFFFFF
2023/01/30 21:15:32 DEBUG : /: >Getattr: errc=0
2023/01/30 21:15:32 DEBUG : /: Releasedir: fh=0x0
2023/01/30 21:15:32 DEBUG : /: >Releasedir: errc=0
2023/01/30 21:15:32 DEBUG : /: Getattr: fh=0xFFFFFFFFFFFFFFFF
2023/01/30 21:15:32 DEBUG : /: >Getattr: errc=0
2023/01/30 21:15:32 DEBUG : /: Opendir:
2023/01/30 21:15:32 DEBUG : /: OpenFile: flags=O_RDONLY, perm=-rwxrwxrwx
2023/01/30 21:15:32 DEBUG : /: >OpenFile: fd=/ (r), err=<nil>
2023/01/30 21:15:32 DEBUG : /: >Opendir: errc=0, fh=0x0
2023/01/30 21:15:32 DEBUG : /: Getpath: Getpath fh=0
2023/01/30 21:15:32 DEBUG : /: >Getpath: errc=0, normalisedPath=""
2023/01/30 21:15:32 DEBUG : /: Releasedir: fh=0x0
2023/01/30 21:15:32 DEBUG : /: >Releasedir: errc=0
2023/01/30 21:15:32 DEBUG : /: Getattr: fh=0xFFFFFFFFFFFFFFFF
2023/01/30 21:15:32 DEBUG : /: >Getattr: errc=0
2023/01/30 21:15:32 DEBUG : /: Getattr: fh=0xFFFFFFFFFFFFFFFF
2023/01/30 21:15:32 DEBUG : /: >Getattr: errc=0
2023/01/30 21:15:32 DEBUG : /: Opendir:
2023/01/30 21:15:32 DEBUG : /: OpenFile: flags=O_RDONLY, perm=-rwxrwxrwx
2023/01/30 21:15:32 DEBUG : /: >OpenFile: fd=/ (r), err=<nil>
2023/01/30 21:15:32 DEBUG : /: >Opendir: errc=0, fh=0x0
2023/01/30 21:15:32 DEBUG : /: Getpath: Getpath fh=0
2023/01/30 21:15:32 DEBUG : /: >Getpath: errc=0, normalisedPath=""
2023/01/30 21:15:32 DEBUG : /: Releasedir: fh=0x0
2023/01/30 21:15:32 DEBUG : /: >Releasedir: errc=0
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
> I have a samba drive with port 9445
>
> It is hosted by Arch Linux samba.

I'd guess this is because it is mis-reporting the block size, or rclone is mis-reading it somehow.

If you use Windows to mount the same smb drive do you see the same thing?

What does this show?

    rclone.exe about mysmbdrive:/Library

**Comment by laggykiller** [comment]
Normal size if I mount on Windows with port 445
```
rclone about mysmbdrive:/Library
Total:   3.609 TiB
Used:    2.619 TiB
Free:    1014.424 GiB
```

**Comment by ncw** [comment]
I think this is a problem with the upstream library and I've reported an issue there: https://github.com/hirochachacha/go-smb2/issues/72

**Comment by Woodie-07** [comment]
I ran into this same issue when trying to use a Hetzner storage box over SMB. The [hirochachacha/go-smb2](https://github.com/hirochachacha/go-smb2) repo hasn't had any updates in over a year, but the [CloudSoda/go-smb2 ](https://github.com/CloudSoda/go-smb2) repo is a fork which gets frequent updates and has [a patch for this issue](https://github.com/CloudSoda/go-smb2/issues/5). After switching rclone to use this fork instead of the original and rebuilding, the issue seems to be fixed (although I have not yet tested it thoroughly).

Before:
![image](https://github.com/rclone/rclone/assets/40068306/a9f4fece-6eb0-4905-80f5-de409104403c)

After:
![image](https://github.com/rclone/rclone/assets/40068306/2525f48d-43cd-44e4-8248-1c773922634d)

**Comment by ncw** [comment]
@Woodie-07 that fork has some nice fixes in it. Its a shame its in a fork though...

I looked through the history and the patches looks sensible to me.

I've decided to switch rclone to this fork as the integration tests all pass with it and it has this fix.

Please give this a test :-)

[v1.65.0-beta.7508.d9d265de0.fix-6733-smb-size](https://beta.rclone.org/branch/fix-6733-smb-size/v1.65.0-beta.7508.d9d265de0.fix-6733-smb-size/) on branch [fix-6733-smb-size](https://github.com/rclone/rclone/tree/fix-6733-smb-size) (uploaded in 15-30 mins)

**Comment by Woodie-07** [comment]
Tested the new beta and all seems to be working, thanks!

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.65
